components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/IoRetry.java

```diff
@@ -0,0 +1,87 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.metrics.reporting.graphite;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+
+/**
+ * Generic retry class for tasks that perform I/O.
+ */
+final class IoRetry {
+
+    @FunctionalInterface
+    public interface IoRunnable {
```

Comment:
Our convention is to put inner classes/interfaces at the bottom of the class. 
Alternatively, did you consider my suggestion of replacing this and `GraphiteReporter.IOAction` with an interface in its own file?